modules/tensor_mechanics/src/auxkernels/RankTwoScalarAux.C

```diff
@@ -47,9 +47,43 @@ RankTwoScalarAux::computeValue()
    case 3:
      val = _tensor[_qp].L2norm();
      break;
+   case 4:
+   case 5:
+   case 6:
+     val = calcEigenValues();
+     break;
    default:
-     mooseError("RankTwoScalarAux Error: Pass valid scalar type - VonMisesStress, EquivalentPlasticStrain, Hydrostatic, L2norm");
+     mooseError("RankTwoScalarAux Error: Pass valid scalar type - VonMisesStress, EquivalentPlasticStrain, Hydrostatic, L2norm MaxPrincipal MidPrincipal MinPrincipal");
   }
   return val;
 }
 
+Real
+RankTwoScalarAux::calcEigenValues()
+{
+  std::vector<Real> eigval(LIBMESH_DIM);
+  Real val;
+  unsigned int max_index = 2;
+  unsigned int mid_index = 1;
+  unsigned int min_index = 0;
+
+  if (LIBMESH_DIM == 2)
+    max_index = 1;
+
+  _tensor[_qp].symmetricEigenvalues(eigval);
+
+  switch (_scalar_type)
+  {
+   case 4:
+     val = eigval[max_index];
+     break;
+   case 5:
+     if (LIBMESH_DIM == 2)
+       mooseError("RankTwoScalarAux Error: No Mid Principal value when LIBMESH_DIM is 2");
+     val = eigval[mid_index];
+     break;
+   case 6:
+     val = eigval[min_index];
+  }
```

Comment:
Please add a default case to handle situations `calcEigenValues()` is called incorrectly, otherwise it will return the uninitialized value `val`.